ethcore/src/block.rs

```diff
@@ -371,6 +380,9 @@ impl<'x> OpenBlock<'x> {
 		let mut s = self;
 
 		s.engine.on_close_block(&mut s.block)?;
+		if s.block.storage_writer.enabled() {
+			s.block.state.write_watched_state(s.block.header_hash, s.block.header.number(), s.block.storage_writer.clone())?;
```

Comment:
That's a performance killer.

Reply:
Totally hear you there, but you only take the hit if the writer is enabled, right? Thinking that it might make sense to have an opt in approach where you take hit on performance in your sync to get significantly improved performance on accessing the data.